eglib/src/gmisc-win32.c

```diff
@@ -116,20 +117,32 @@ g_path_is_absolute (const char *filename)
 const gchar *
 g_get_home_dir (void)
 {
-	/* FIXME */
-	const gchar *drive = g_getenv ("HOMEDRIVE");
-	const gchar *path = g_getenv ("HOMEPATH");
 	gchar *home_dir = NULL;
-	
-	if (drive && path) {
-		home_dir = g_malloc(strlen(drive) + strlen(path) +1);
-		if (home_dir) {
-			sprintf(home_dir, "%s%s", drive, path);
-		}
+
+	PWSTR profile_path = NULL;
+	HRESULT hr = SHGetKnownFolderPath (&FOLDERID_Profile, KF_FLAG_DEFAULT, NULL, &profile_path);
```

Comment:
Looks like mingw doesn't know SHGetKnownFolderPath (or we somehow end up not defining _WIN32_WINNT=0x0600 in eglib):

```
gmisc-win32.c: In function 'monoeg_g_get_home_dir':
gmisc-win32.c:123:2: warning: implicit declaration of function 'SHGetKnownFolderPath' [-Wimplicit-function-declaration]
  HRESULT hr = SHGetKnownFolderPath (&FOLDERID_Profile, KF_FLAG_DEFAULT, NULL, &profile_path);
  ^
gmisc-win32.c:123:56: error: 'KF_FLAG_DEFAULT' undeclared (first use in this function)
  HRESULT hr = SHGetKnownFolderPath (&FOLDERID_Profile, KF_FLAG_DEFAULT, NULL, &profile_path);
```

What we've done in the past is simply including the declaration ourselves wrapped in `#if !defined`